packages/m2m-gateway/src/services/clientService.ts

```
@@ -1,8 +1,30 @@
+import { ClientId, UserId, unsafeBrandId } from "pagopa-interop-models";
 import { PagoPAInteropBeClients } from "../clients/clientsProvider.js";
+import { M2MGatewayAppContext } from "../utils/context.js";
 
 export type ClientService = ReturnType<typeof clientServiceBuilder>;
 
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
-export function clientServiceBuilder(_clients: PagoPAInteropBeClients) {
-  return {};
+export function clientServiceBuilder(clients: PagoPAInteropBeClients) {
+  return {
+    /**
+     * Get the admin user ID for a given client ID.
+     * This method is used by the m2mAuthDataValidationMiddleware to validate
+     * that the userId in the JWT token is the same as the adminId associated
+     * with the client associated with the token.
+     */
+    async getClientAdminId(
+      clientId: ClientId,
+      headers: M2MGatewayAppContext["headers"]
+    ): Promise<UserId | undefined> {
+      const client = await clients.authorizationClient.client.getClient({
+        params: { clientId },
+        headers,
+      });
+
+      return client.data.adminId
+        ? unsafeBrandId<UserId>(client.data.adminId)
+        : undefined;
```

Could you clarify the reasoning behind returning `undefined` in this case, instead of throwing an error or handling it differently?

---

No speficific reason, actually it's better to throw an error as we usually do in services business logic in these cases, fixed in https://github.com/pagopa/interop-be-monorepo/pull/1840/commits/39672ace3d6f13ae9798afdfe1ff9bc686e610c8
Thanks 🙏